src/Shared/RoslynUtils/WellKnownTypes.cs

```diff
@@ -86,6 +86,26 @@ private INamedTypeSymbol GetAndCache(int index)
         return _lazyWellKnownTypes[index]!;
     }
 
+    // Look for types within well-known assemblies only.
+    private INamedTypeSymbol? GetTypeByMetadataNameInTargetAssembly(string metadataName)
+    {
+        var types = _compilation.GetTypesByMetadataName(metadataName);
+        if (types.Length == 1)
+        {
+            return types[0];
+        }
+
+        foreach (var type in types)
+        {
+            if (type.ContainingAssembly.Identity.Name.StartsWith("System", StringComparison.Ordinal)
+                || type.ContainingAssembly.Identity.Name.StartsWith("Microsoft", StringComparison.Ordinal))
```

Comment:
Nit: this will accept `MicrosoftAlternative`.

Reply:
Fixed.